### PR TITLE
Unstuctured storage variable hash strings and renamings

### LINKED ISSUE #206: Precompute unstructured storage hashes
Depends on #181 and #204.

See https://github.com/merklejerk/solpp.

### PR DESCRIPTION
**Pull request by lxzrv**
Closes #206

### CODE REVIEW

File: contracts/0.4.24/Lido.sol [hunk 1]
```diff
@@ -51,28 +51,28 @@ contract Lido is ILido, IsContract, Pausable, AragonApp {
...
+    bytes32 internal constant TREASURY_FEE_VALUE_POSITION = keccak256("treasuryFee.lido.lido.eth");
+    bytes32 internal constant INSURANCE_FEE_VALUE_POSITION = keccak256("insuranceFee.lido.lido.eth");
+    bytes32 internal constant NODE_OPERATORS_FEE_VALUE_POSITION = keccak256("operatorsFee.lido.lido.eth");
 
-    bytes32 internal constant TOKEN_VALUE_POSITION = keccak256("lido.Lido.token");
-    bytes32 internal constant VALIDATOR_REGISTRATION_VALUE_POSITION = keccak256("lido.Lido.validatorRegistration");
-    bytes32 internal constant ORACLE_VALUE_POSITION = keccak256("lido.Lido.oracle");
-    bytes32 internal constant NODE_OPERATOR_REGISTRY_VALUE_POSITION = keccak256("lido.Lido.nodeOperatorRegistry");
+    bytes32 internal constant TOKEN_VALUE_POSITION = keccak256("token.lido.lido.eth");
+    bytes32 internal constant VALIDATOR_REGISTRATION_VALUE_POSITION = keccak256("validatorRegistration.lido.lido.eth");
```
Comment (ongrid) [inline contracts/0.4.24/Lido.sol:60]: rename `validatorRegistration` to `depositContract`, Everywhere. It's the old deprecated name.

File: contracts/0.4.24/Lido.sol [hunk 2]
```diff
@@ -51,28 +51,28 @@ contract Lido is ILido, IsContract, Pausable, AragonApp {
...
     uint256 internal constant DEFAULT_MAX_DEPOSITS_PER_CALL = 16;
 
-    bytes32 internal constant FEE_VALUE_POSITION = keccak256("lido.Lido.fee");
-    bytes32 internal constant TREASURY_FEE_VALUE_POSITION = keccak256("lido.Lido.treasuryFee");
-    bytes32 internal constant INSURANCE_FEE_VALUE_POSITION = keccak256("lido.Lido.insuranceFee");
-    bytes32 internal constant NODE_OPERATORS_FEE_VALUE_POSITION = keccak256("lido.Lido.operatorsFee");
+    bytes32 internal constant FEE_VALUE_POSITION = keccak256("fee.lido.lido.eth");
+    bytes32 internal constant TREASURY_FEE_VALUE_POSITION = keccak256("treasuryFee.lido.lido.eth");
+    bytes32 internal constant INSURANCE_FEE_VALUE_POSITION = keccak256("insuranceFee.lido.lido.eth");
+    bytes32 internal constant NODE_OPERATORS_FEE_VALUE_POSITION = keccak256("operatorsFee.lido.lido.eth");
```
Comment (ongrid) [inline contracts/0.4.24/Lido.sol:57]: nodeOperatorsFee

File: contracts/0.4.24/Lido.sol (hunk 1 above)
Comment (ongrid) [reply contracts/0.4.24/Lido.sol:60]: Do this with separate commit

**Review by ongrid — CHANGES_REQUESTED** [review]
Pausable.sol still contains old notation
NodeOperatorsRegistry.sol not touched at all

File: apps/lido/app/src/App.js [hunk 3]
```diff
@@ -83,7 +83,7 @@ export default function App() {
       bufferedEther,
       totalPooledEther,
       token,
-      validatorRegistrationContract,
+      depositContractContract,
```
Comment (ongrid) [inline apps/lido/app/src/App.js:86]: just depositContract

File: apps/lido/app/src/App.js [hunk 4]
```diff
@@ -238,8 +238,8 @@ export default function App() {
         content: <strong>{formatEth(totalPooledEther) || 'No data'}</strong>,
       },
       {
-        label: 'Validator Registration Contract',
-        content: <IdentityBadge entity={validatorRegistrationContract} />,
+        label: 'Deposit contract Contract',
```
Comment (ongrid) [inline apps/lido/app/src/App.js:241]: just depositContract

File: apps/lido/app/src/App.js [hunk 5]
```diff
@@ -238,8 +238,8 @@ export default function App() {
...
+        content: <IdentityBadge entity={depositContractContract} />,
```
Comment (ongrid) [inline apps/lido/app/src/App.js:242]: just depositContract

File: contracts/0.4.24/Lido.sol [hunk 6]
```diff
@@ -45,34 +45,34 @@ contract Lido is ILido, IsContract, Pausable, AragonApp {
 
     uint256 constant public DEPOSIT_SIZE = 32 ether;
 
-    uint256 internal constant MIN_DEPOSIT_AMOUNT = 1 ether;     // validator_registration.vy
-    uint256 internal constant DEPOSIT_AMOUNT_UNIT = 1000000000 wei;     // validator_registration.vy
+    uint256 internal constant MIN_DEPOSIT_AMOUNT = 1 ether;     // deposit_contract.vy
```
Comment (ongrid) [inline contracts/0.4.24/Lido.sol:48]: the comment // deposit_contract.vy is irrelevant here. Remove it

File: contracts/0.4.24/Lido.sol [hunk 7]
```diff
@@ -45,34 +45,34 @@ contract Lido is ILido, IsContract, Pausable, AragonApp {
...
+    uint256 internal constant DEPOSIT_AMOUNT_UNIT = 1000000000 wei;     // deposit_contract.vy
```
Comment (ongrid) [inline contracts/0.4.24/Lido.sol:49]: the comment // deposit_contract.vy is irrelevant here. Remove it

File: docs/protocol-levers.md [hunk 8]
```diff
@@ -175,7 +175,7 @@ is sent to node operators’ reward addresses (`_rewardAddress`).
   * Permission required: `SET_NODE_OPERATOR_ACTIVE_ROLE`
 
 Misbehaving node operators can be deactivated by calling this function. The pool skips
-deactivated operators during validator registration; also, deactivated operators don’t
+deactivated operators during deposit contract; also, deactivated operators don’t
```
Comment (ongrid) [inline docs/protocol-levers.md:178]: This shouldn't be changed. See the context, it's not about deposit contract.

File: test/scenario/lido_happy_path.js [hunk 9]
```diff
@@ -164,12 +164,12 @@ contract('Lido: happy path', (addresses) => {
     await web3.eth.sendTransaction({ to: pool.address, from: user2, value: ETH(30) })
     await pool.depositBufferedEther()
 
-    // The first 32 ETH chunk was deposited to the validator registration contract,
+    // The first 32 ETH chunk was deposited to the deposit contract contract,
```
Comment (ongrid) [inline test/scenario/lido_happy_path.js:167]: just "deposit contract"

**Review by ongrid — CHANGES_REQUESTED** [review]
1. need to rebase to the HEAD of origin/pre-mainnet-refactoring (coordinated with @skozin to use pre-mainnet-refactoring as the develop branch)
2. All mentioning of deposit_contract.vy should be renamed to deposit_contract.sol (has been migrated from Vyper to Solidity). Irrelevant comments should be removed, I've highlighted them
3. A lot of cosmetics, read the resulting text before

**Review by ongrid — APPROVED** [review]

**Review by skozin — APPROVED** [review]

**Pull request merged**